docs/content/guides/dagster-pipes/modifying-existing-code-to-work-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,270 @@
 ---
 title: "Modifying existing code to work with Dagster Pipes | Dagster Docs"
-description: ""
+description: "With Dagster Pipes, you can incorporate existing code into Dagster without huge refactors. This guide shows you how to modify existing code to work with Dagster Pipes."
 ---
 
 # Modifying existing code to work with Dagster Pipes
+
+In this guide, we’ll demonstrate how to modify existing code (e.g. an existing Python script) to work with [Dagster Pipes](/guides/dagster-pipes) and send metadata from the script to Dagster.
+
+<Note>
+
+This guide is specific to modifying existing code that is being orchestrated by Dagster. Looking for other uses of Dagster Pipes? Refer to the <a href="/guides/dagster-pipes#usage">Dagster Pipes Usage</a> documentation.
+
+</Note>
+
+## Prerequisites
+
+<Note>
+
+This guide assumes your code is being orchestrated by Dagster. If that’s not the case, please refer to <a href="integrating-subprocess-with-dagster-pipes">Integrating Dagster Pipes with Subprocess</a> for how to orchestrate your existing Python scripts using Dagster.
```

Comment:
To me this line is a little confusing because of the word "orchestrate". I think it would be clearer to say: 

"This guide talks about how to modify your code to send information back to Dagster. To schedule and automate this code through Dagster you will need to follow this guide ... "

Basically, I think of it as a two step process to get Dagster to run some existing code:

1. You write some Dagster code (defining the assets & adding the schedules/sensors/etc to trigger it)
2. You modify the existing code
3. (optonal) you modify the existing code further if you want additional metadata sent back to Dagster

To me this checklist is currently hard to understand because it is split across guides, and the cross-refs don't really make it clear you need to do both steps.

Reply:
I went back end forth on the sequence of pages a lot! I think you are exactly right about the two step process. The current split is indeed confusing because the two pages are kinda depend on each other - maybe, a better approach is to structure the guides to be similar to the [dbt + dagster tutorial](https://docs.dagster.io/integrations/dbt/using-dbt-with-dagster), where we rearrange this and https://github.com/dagster-io/dagster/pull/17096 to be 3 separate pages:
- Part 1: Create Dagster assets (for certain stakeholders, if their code is already being orchestrated by some asset factory set up by their data platform folks, they can skip Part 1 and go directly to Part 2)
- Part 2: Modify external code
- Part 3: Send info back to Dagster

@erinkcochran87 what do you think?

Reply:
@slopp i'm experimenting a new structure for the subprocess docs here: https://github.com/dagster-io/dagster/pull/17108 - please let me know how you feel about the proposed new structure.